js_modules/dagster-ui/packages/ui-core/src/asset-data/AssetLiveDataProvider.tsx

```diff
@@ -1,172 +1,91 @@
-    let didUpdateOnce = false;
-    let didScheduleUpdateOnce = false;
-    let updates: {stringKey: string; assetData: LiveDataForNode | undefined}[] = [];
-
-    function processUpdates() {
-      setData((data) => {
-        const copy = {...data};
-        updates.forEach(({stringKey, assetData}) => {
-          if (assetData) {
-            copy[stringKey] = assetData;
-          } else {
-            delete copy[stringKey];
-          }
-        });
-        updates = [];
-        return copy;
-      });
-    }
-
-    const setDataSingle = (stringKey: string, assetData?: LiveDataForNode) => {
-      /**
-       * Throttle updates to avoid triggering too many GCs and too many updates when fetching 1,000 assets,
-       */
-      updates.push({stringKey, assetData});
-      if (!didUpdateOnce) {
-        if (!didScheduleUpdateOnce) {
-          didScheduleUpdateOnce = true;
-          requestAnimationFrame(() => {
-            processUpdates();
-            didUpdateOnce = true;
-          });
-        }
-      } else if (!timeout) {
-        timeout = setTimeout(() => {
-          processUpdates();
-          timeout = null;
-        }, batchUpdatesInterval);
-      }
-    };
-    const unsubscribeCallbacks = assetKeys.map((key) =>
-      manager.subscribe(key, setDataSingle, thread),
-    );
-    return () => {
-      unsubscribeCallbacks.forEach((cb) => {
-        cb();
-      });
-    };
-  }, [assetKeys, batchUpdatesInterval, manager, thread]);
-
-  return {
-    liveDataByNode: data,
-
-    refresh: React.useCallback(() => {
-      manager.refreshKeys(assetKeys);
-      setIsRefreshing(true);
-    }, [assetKeys, manager]),
-
-    refreshing: React.useMemo(() => {
-      if (isRefreshing && !manager.areKeysRefreshing(assetKeys)) {
-        setTimeout(() => {
-          setIsRefreshing(false);
-        });
-        return false;
-      }
-      return true;
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [assetKeys, data, isRefreshing]),
-  };
+  return factory.useLiveData(
+    assetKeys.map((key) => tokenForAssetKey(key)),
+    thread,
+  );
 }
 
 export const AssetLiveDataProvider = ({children}: {children: React.ReactNode}) => {
   const [allObservedKeys, setAllObservedKeys] = React.useState<AssetKeyInput[]>([]);
 
-  const client = useApolloClient();
-  const manager = AssetLiveDataThreadManager.getInstance(client);
-
-  const [isGloballyRefreshing, setIsGloballyRefreshing] = React.useState(false);
-  const [oldestDataTimestamp, setOldestDataTimestamp] = React.useState(0);
-
-  const onUpdatingOrUpdated = React.useCallback(() => {
-    const {isRefreshing, oldestDataTimestamp} = manager.getOldestDataTimestamp();
-    setIsGloballyRefreshing(isRefreshing);
-    setOldestDataTimestamp(oldestDataTimestamp);
-  }, [manager]);
-
   React.useEffect(() => {
-    manager.setOnSubscriptionsChangedCallback((keys) =>
+    factory.manager.setOnSubscriptionsChangedCallback((keys) =>
```

Comment:
Just in terms of naming conventions this feels a bit odd - the factory could build and return a manager, or the factory and the manager could be the same object maybe. If the factory is a singleton why have another nested singleton manager inside it? 

Maybe the manager could have a few static methods, which set up instances for different use cases?